crates/uv-python/src/discovery.rs

```diff
@@ -410,6 +413,65 @@ fn python_executables_from_search_path<'a>(
         })
 }
 
+/// Find all acceptable `python3.x` minor versions.
+///
+/// For example, let's say `python` and `python3` are Python 3.10. When a user requests `>= 3.11`,
+/// we still need to find a `python3.12` in PATH.
+fn find_all_minor(
+    implementation: Option<&ImplementationName>,
+    version_request: &VersionRequest,
+    dir_clone: &Path,
+) -> impl Iterator<Item = PathBuf> {
+    match version_request {
+        VersionRequest::Any | VersionRequest::Major(_) | VersionRequest::Range(_) => {
+            let regex = if let Some(implementation) = implementation {
+                Regex::new(&format!(
+                    r"^({}|python3)\.\d\d?{}$",
+                    regex::escape(&implementation.to_string()),
+                    regex::escape(EXE_SUFFIX)
+                ))
+                .unwrap()
+            } else {
+                Regex::new(&format!(r"^python3\.\d\d?{}$", regex::escape(EXE_SUFFIX))).unwrap()
+            };
+            let all_minors = which::which_re_in(&regex, Some(&dir_clone))
+                .into_iter()
+                .flatten()
+                .filter(move |path| {
+                    // Filter out interpreter we already know have a too low minor version.
+                    let minor = path
+                        .file_name()
+                        .and_then(|filename| filename.to_str())
```

Comment:
I feel like we should just not use `which` and we should iterate over the executables in the directory manually. A regex feels heavier than it needs to be. Wdyt?

Reply:
That's an implementation i used to have (without `which` it's much better than the current one), but it failed because `which` doesn't expose `Checker` so we'd have to vendor that code from them to check whether we can use a specific file.

Reply:
I'm okay with vendoring that code if you are.

Reply:
Added `is_executable`